### ISSUE TITLE
error in running reconstruct_contig.py

### ISSUE THREAD

**Issue by tengdiyouying608** [Magdoll/Cogent#28]
Hi:
   It reported ValueError when I ran reconstruct_contig.py with the following messages:
Traceback (most recent call last):
  File "/share/nas1/liyingying/tools/soft/cogent/Anaconda/anaconda3/envs/anaCogent/bin/reconstruct_contig.py", line 4, in <module>
    __import__('pkg_resources').run_script('Cogent==2.1', 'reconstruct_contig.py')
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/pkg_resources/__init__.py", line 750, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1527, in run_script
    exec(code, namespace, namespace)
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/Cogent-2.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 352, in <module>
    main()
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/Cogent-2.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 268, in main
    run_Cogent_on_split_files(dirs, depth=0)
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/Cogent-2.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 71, in run_Cogent_on_split_files
    run_Cogent_on_input()
  File "/cogent/Anaconda/anaconda3/envs/anaCogent/lib/python2.7/site-packages/Cogent-2.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 157, in run_Cogent_on_input
    seqid, weight = line.strip().split('\t')
ValueError: need more than 1 value to unpack

I have no idea to for this.Could you help me ?

Thanks!

LIYY

**Comment by Magdoll** [comment]
Hi @tengdiyouying608 ,

It looks like in the split directories, the file `in.weights` is not being properly propagated.

The `in.weights` file should look like below, where one column is the sequence ID and the second column is the weight (default is 1). The two columns should be separated by a tab. 

```
i2_HQ_samplee5d47c|c63386/f3p36/2608    1
i2_HQ_samplee5d47c|c7469/f2p32/2292 1
i2_HQ_samplee46251|c31503/f2p29/2286    1
```

Can you check that you have `in.weights` in the run directory and the split directories?

--Liz

**Comment by angeladarling** [comment]
Hi:
It reported ValueError when I ran reconstruct_contig.py  --nx_cycle_detection -k 20 10603_1 -p 10603_1  with the following messages:

Traceback (most recent call last):
  File "reconstruct_contig.py", line 4, in <module>
    __import__('pkg_resources').run_script('Cogent==3.1', 'reconstruct_contig.py')
  File "/share/public/software/WDLIB/lib/FALCON_unzip_0619.2017/lib/python2.7/site-packages/pkg_resources/__init__.py", line 739, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/share/public/software/WDLIB/lib/FALCON_unzip_0619.2017/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1494, in run_script
    exec(code, namespace, namespace)
  File "/share/work2/liuying/software/anaconda2/envs/anaCogent/lib/python2.7/site-packages/Cogent-3.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 337, in <module>
    main()
  File "/share/work2/liuying/software/anaconda2/envs/anaCogent/lib/python2.7/site-packages/Cogent-3.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 259, in main
    run_Cogent_on_input()
  File "/share/work2/liuying/software/anaconda2/envs/anaCogent/lib/python2.7/site-packages/Cogent-3.1-py2.7.egg/EGG-INFO/scripts/reconstruct_contig.py", line 218, in run_Cogent_on_input
    sp.contract_sinks(G, path_d, mermap)
  File "/share/work2/liuying/software/anaconda2/envs/anaCogent/lib/python2.7/site-packages/Cogent-3.1-py2.7.egg/Cogent/splice_graph.py", line 170, in contract_sinks
    assert path_d[k][-1] == sink and path_d[k][-2] == pred
AssertionError


I have no idea to for this.Could you help me ? 
Thanks! 
--

**Comment by Magdoll** [comment]
Hi @angeladarling ,

It looks like you are posting about a completely different issue. Please file through a new "issue" instead of using an existing unrelated issue.

Also, some of the bugs you see in 3.1 may be fixed in v3.3. Please update the v3.3 and try again. If the error remains, I can look at the input data for you.

Thanks,

--Liz
